[TP][EZ] Update doc for TP parallel style

fduwjj opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #107819


We need to update the doc for PairwiseParallel and SequenceParallel so that users don't get wrong impressions that these working for ``nn.Transformer``.

fduwjj committed:
[TP] Update doc for parallel style

[ghstack-poisoned]

awgu commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -51,8 +51,10 @@ class PairwiseParallel(ParallelStyle):
     .. warning::
-        PairwiseParallel only supports ``nn.Multihead Attention``,
-        ``nn.Transformer`` or even-number-layer MLP for now.
+        PairwiseParallel does not support ``nn.Multihead Attention``,
```
```suggestion
        PairwiseParallel does not support ``nn.MultiheadAttention``,
```

awgu commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -51,8 +51,10 @@ class PairwiseParallel(ParallelStyle):
-        ``nn.Transformer`` or even-number-layer MLP for now.
+        PairwiseParallel does not support ``nn.Multihead Attention``,
+        ``nn.Transformer`` well at this moment. One workaround is to
+        use `ColwiseParallel` and `RowwiseParallel` directly. We recommend to
```
```suggestion
        use ``ColwiseParallel`` and ``RowwiseParallel`` directly. We recommend to
```

awgu commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -51,8 +51,10 @@ class PairwiseParallel(ParallelStyle):
+        PairwiseParallel does not support ``nn.Multihead Attention``,
+        ``nn.Transformer`` well at this moment. One workaround is to
+        use `ColwiseParallel` and `RowwiseParallel` directly. We recommend to
+        use `PairwiseParallel` only for even-number-layer MLP for now.
```
```suggestion
        use ``PairwiseParallel`` only for even-number-layer MLP for now.
```

awgu commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -73,8 +75,10 @@ class SequenceParallel(PairwiseParallel):
     We assume both input and output need to be sharded DTensors.
 
     .. warning::
```
Same nits as above

awgu reviewed (approved):
I think that single backtick will be italics instead of code type.

fduwjj committed:
Update torch/distributed/tensor/parallel/style.py

Co-authored-by: Andrew Gu <31054793+awgu@users.noreply.github.com>

fduwjj committed:
Update torch/distributed/tensor/parallel/style.py

Co-authored-by: Andrew Gu <31054793+awgu@users.noreply.github.com>

fduwjj committed:
Update torch/distributed/tensor/parallel/style.py

Co-authored-by: Andrew Gu <31054793+awgu@users.noreply.github.com>

fduwjj committed:
Update on "[TP][EZ] Update doc for TP parallel style"



We need to update the doc for PairwiseParallel and SequenceParallel so that users don't get wrong impressions that these working for ``nn.Transformer``.



[ghstack-poisoned]

wanchaol commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -51,8 +51,10 @@ class PairwiseParallel(ParallelStyle):
-        ``nn.Transformer`` or even-number-layer MLP for now.
+        PairwiseParallel does not support ``nn.MultiheadAttention``,
+        ``nn.Transformer`` well at this moment. One workaround is to
+        use ``ColwiseParallel`` and ``RowwiseParallel`` directly. We recommend to
```
nit: One workaround is to apply ``ColwiseParallel`` and ``RowwiseParallel`` to the components of transformer directly.

wanchaol commented on torch/distributed/tensor/parallel/style.py:
```diff
@@ -73,8 +75,10 @@ class SequenceParallel(PairwiseParallel):
-        ``nn.Transformer`` or even-number-layer MLP for now.
+        SequenceParallel does not support ``nn.MultiheadAttention``,
+        ``nn.Transformer`` well at this moment. One workaround is to
+        use ``ColwiseParallel`` and ``RowwiseParallel`` directly. We recommend to
```
ditto

wanchaol reviewed (approved):
lgtm

fduwjj committed:
Update on "[TP][EZ] Update doc for TP parallel style"



We need to update the doc for PairwiseParallel and SequenceParallel so that users don't get wrong impressions that these working for ``nn.Transformer``.



[ghstack-poisoned]

fduwjj commented:
@pytorchbot merge
